migrate_rules.py

```diff
@@ -64,62 +64,196 @@ def migrate(rancher_url, rancher_api_token, cluster_id, project_id, insecure):
+                        })
+                    ]
                 }
-            ]
+            }
+            if "description" in alert_group:
+                prometheus_rule["metadata"]["annotations"]["field.cattle.io/description"] = alert_group["description"]
+
+            prometheus_rules.append(prometheus_rule)
+
+            if "recipients" in alert_group:
+                for recipient in alert_group["recipients"]:
+                    notifier_id = recipient["notifierId"]
+                    notifiers_by_id[notifier_id]["group_ids"].append(alert_group["id"])
+
+    alertmanager_yaml = update_alertmanager_config(alertmanager_yaml, notifiers_by_id)
+
+    print(yaml.dump(alertmanager_yaml), end="---\n")
+
+    # Create resources
+    namespace = OrderedDict(**{
+        "apiVersion": "v1",
+        "kind": "Namespace",
+        "metadata": {
+            "name": "cattle-monitoring-system"
         }
-    }
+    })
+
+    alerting_config = OrderedDict(**{
+        "apiVersion": "v1",
+        "kind": "Secret",
+        "metadata": {
+            "name": "alertmanager-rancher-monitoring-alertmanager",
+            "namespace": "cattle-monitoring-system",
+            "labels": {
+                "source": "rancher-alerting-v1"
+            },
+        },
+        "data": {
+            "alertmanager.yaml": literal(base64_encode(yaml.dump(alertmanager_yaml))),
+            "notification.tmpl": literal(alerting_config["data"]["notification.tmpl"])
+        },
+        "type": "Opaque"
+    })
+    
+    resources = [namespace, alerting_config]
+
+    resources.extend(prometheus_rules)
 
-    print(yaml.dump(prometheus_rules))
+    print(yaml.dump_all(resources))
+
+def update_alertmanager_config(alertmanager_yaml, notifiers_by_id):
+    alertmanager_yaml["route"]["receiver"] = "null"
+    alertmanager_yaml["route"]["group_by"] = ['job']
+    alertmanager_yaml["templates"] = ["/etc/alertmanager/config/*.tmpl"]
+    
+    # Update to one Receiever per Notifier instead of one per alert group
+    receiver_by_group_id = {}
+    receivers = [{"name": "null"}]
+    for notifier_id in notifiers_by_id:
+        notifier = notifiers_by_id[notifier_id]
+        if len(notifier["group_ids"]) == 0:
+            continue
+        # Get receiver configuration
+        first_matching_notifier = notifier["group_ids"][0]
+        receiver = [r for r in alertmanager_yaml["receivers"] if r["name"] == first_matching_notifier][0]
+        receiver["name"] = notifier["name"]
+        receivers.append(receiver)
+        # Keep track of what group_ids need to be modified
+        for group_id in notifier["group_ids"]:
+            receiver_by_group_id[group_id] = receiver["name"]
+    alertmanager_yaml["receivers"] = receivers
+    
+    # Update Recievers attached to routes accordingly
+    alertmanager_yaml["route"]["receiver"] = "null"
+    alertmanager_yaml["route"]["group_by"] = ['job']
+    routes = []
+    for route in alertmanager_yaml["route"]["routes"]:
```

Comment:
I think the generated config should still contain the default route to explicitely send the Watchdog alert to the null receiver:
```
      routes:
      - match:
          alertname: Watchdog
        receiver: 'null'
```

Reply:
Sounds good, will add it to the PR.